api/filter_dimensions.go

```diff
@@ -438,18 +438,37 @@ func createPublicDimension(dimension models.Dimension, host, filterID string) *m
 	return publicDim
 }
 
-func removeDuplicateAndEmptyOptions(elements []string) []string {
+func RemoveDuplicateAndEmptyOptions(elements []string) []string {
 	encountered := map[string]bool{}
 	result := []string{}
 
-	for v := range elements {
-		if !encountered[elements[v]] {
-			encountered[elements[v]] = true
-			if elements[v] != "" {
-				result = append(result, elements[v])
+	for _, v3 := range elements {
+		if !encountered[v3] {
+			encountered[v3] = true
+			if v3 != "" {
+				result = append(result, v3)
 			}
 		}
 	}
 
 	return result
 }
+
+func getStringArrayFromInterface(elements interface{}) []string {
```

Comment:
I would suggest that this function returns ([]string, error), returning an error in any `!ok` case.

Reply:
Thanks for review.
Improvement applied.